editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -291,14 +292,14 @@ void TileMapEditorTilesPlugin::_update_source_display() {
 	}
 }
 
-void TileMapEditorTilesPlugin::_patterns_item_list_gui_input(const Ref<InputEvent> &p_event) {
-	TileMap *tile_map = Object::cast_to<TileMap>(ObjectDB::get_instance(tile_map_id));
-	if (!tile_map) {
+void TileMapLayerEditorTilesPlugin::_patterns_item_list_gui_input(const Ref<InputEvent> &p_event) {
+	TileMapLayer *edited_layer = Object::cast_to<TileMapLayer>(ObjectDB::get_instance(edited_tile_map_layer_id));
```

Comment:
I think this deserves a helper method, because it's done everywhere.

Reply:
You mean this part  `Object::cast_to<TileMapLayer>(ObjectDB::get_instance(edited_tile_map_layer_id));` ?
I know this this snippet is used multiple time but I am not sure I can squash it in a helper while keeping things readable:
```
	TileMapLayer *edited_layer = Object::cast_to<TileMapLayer>(ObjectDB::get_instance(edited_tile_map_layer_id));
	if (!edited_layer) {
		return;
	}

	Ref<TileSet> tile_set = edited_layer->get_effective_tile_set();
	if (tile_set.is_null()) {
		return;
	}
```

Reply:
Well it could be something like this
```
TileMapLayer *edited_layer = get_edited_layer();
```
Not sure how it can negatively impact readability.

I'm not referring to the TileSet part, as it's not used everywhere.

Reply:
> Not sure how it can negatively impact readability.

I meant if I had to compact the whole snippet. But just for the edited_layer that makes sense. I'll do that then.